falcon/media/multipart.py

```diff
@@ -0,0 +1,387 @@
+                value = self._headers.get(b'content-disposition', b'')
+                self._content_disposition = cgi.parse_header(value.decode())
+
+            _, params = self._content_disposition
+
+            # NOTE(vytas): Supporting filename* as per RFC5987, as that has
+            #   been spotted in the wild, even though RFC 7578 forbids it.
+            match = _FILENAME_STAR_RFC5987.match(params.get('filename*', ''))
+            if match:
+                charset, value = match.groups()
+                charset = charset.lower()
+                if charset not in self._parse_options.supported_charsets:
+                    raise MultipartParseError(
+                        'unsupported charset: {}'.format(charset))
+                self._filename = unquote_to_bytes(value).decode(charset)
+            else:
+                value = params.get('filename')
+                if value is None:
+                    return None
+                self._filename = value
+
+        return self._filename
+
+    @property
+    def secure_filename(self):
+        return misc.secure_filename(self.filename)
+
+    @property
+    def name(self):
+        if self._name is None:
+
+            if self._content_disposition is None:
+                value = self._headers.get(b'content-disposition', b'')
+                self._content_disposition = cgi.parse_header(value.decode())
+
+            _, params = self._content_disposition
+            self._name = params.get('name')
+
+        return self._name
+
+    @property
+    def media(self):
+        if self._media is None:
+            handler = self._parse_options.media_handlers.find_by_media_type(
+                self.content_type, 'text/plain')
+            self._media = handler.deserialize(
+                self.stream, self.content_type, None)
+
+        return self._media
+
+
+class MultipartForm:
+
+    def __init__(self, stream, boundary, content_length, parse_options):
+        # if not isinstance(stream, BufferedStream):
+        if not hasattr(stream, 'read_until'):
+            if isinstance(stream, request_helpers.BoundedStream):
+                stream = BufferedStream(stream.stream.read, content_length)
+            else:
+                stream = BufferedStream(stream.read, content_length)
+
+        self._stream = stream
+        self._boundary = boundary
+        self._dash_boundary = b'--' + boundary
+        self._parse_options = parse_options
+
+    def __iter__(self):
+        delimiter = self._dash_boundary
+        stream = self._stream
+        max_headers_size = self._parse_options.max_body_part_headers_size
+        remaining_part_count = self._parse_options.max_body_part_count
+
+        while True:
+            # NOTE(vytas): Either exhaust the unused stream part, or skip
+            #   the prologue.
+            stream.pipe_until(delimiter)
+            stream.read(len(delimiter))
+
+            if not delimiter.startswith(b'\r\n'):
```

Comment:
Instead of having b`'\r\n'` all over the place, could you make "constant" `CLRF` at the top of the file and use that? It's three less characters you need to type :)

Reply:
Sure, thanks, can do that :+1: (I would go with the more common abbreviation `CRLF` though).

Reply:
Whoops! `CRLF` is correct... apparently I can't type straight.